### PR TITLE
Restructured omsicpp system library to reduce cpp fmu size

### PR DESCRIPTION
**Pull request by niklwors**

### CODE REVIEW

File: OMCompiler/SimulationRuntime/OMSICpp/runtime/include/Core/System/ExtendedSystem.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Implementation of standart functions (e.g. giveRHS(...), etc.).
+Provision of member variables used by all systems.
+
+Note:
+The order of variables in the extended state vector perserved (see: "Sorting
+variables by using the index" in "Design proposal for a general solver interface
+for Open Modelica", September, 10 th, 2008
+
+
+\date     October, 1st, 2008
```
Comment (AnHeuermann) [inline OMCompiler/SimulationRuntime/OMSICpp/runtime/include/Core/System/ExtendedSystem.h:19]: Is this an old file? Otherwise we should update the date.

**Comment by AnHeuermann** [comment]
Can you point out what is actually changed?
As far as I can see you changed the OMSI libraries and split them up in multiple ones. So a better linking for FMUs only links to the needed ones?

File: OMCompiler/SimulationRuntime/OMSICpp/runtime/include/Core/System/ExtendedSystem.h (hunk 1 above)
Comment (niklwors) [reply OMCompiler/SimulationRuntime/OMSICpp/runtime/include/Core/System/ExtendedSystem.h:19]: No this is a new class, I will update it.

Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/OMSICpp/runtime/include/Core/System/ExtendedSystem.h:19]: There are a few other places with the same stuff.

**Comment by niklwors** [comment]
> 
> 
> Can you point out what is actually changed?
> As far as I can see you changed the OMSI libraries and split them up in multiple ones. So a better linking for FMUs only links to the needed ones?

yes this is correct. Before we had one system library with basic functionalities. Now there are three system libraries. For an FMU only a smaller system has to be linked now

**Comment by niklwors** [comment]
@adrpo  Do you know why the build doesn't work? I can't see the error.

**Comment by casella** [comment]
@adrpo, can you merge this PR?

**Pull request merged**